lib/generation/generators/util/pb_generation_view_data.dart

```diff
@@ -77,11 +77,22 @@ class PBGenerationViewData {
       _globalVariables.addAll(variable);
 
   void addMethodVariable(PBVariable variable) {
-    if (!_isDataLocked && variable != null) {
+    if (!_isDataLocked && variable != null && !_contains(variable)) {
       _methodVariables.add(variable);
     }
   }
 
+  bool _contains(PBVariable variable) {
+    var result = false;
+    _methodVariables.forEach((element) {
+      if (element.variableName == variable.variableName) {
+        result = true;
+        return result;
+      }
+    });
+    return result;
+  }
+
```

Comment:
I would suggest you use a Map or Set, probably won't matter too much for the amount of variables in this case, but it's good practice to reduce the search complexity to O(log n).  The underlying binary search tree for these items uses a red-black tree.

Reply:
Good point. @ivan-015 @SushiRoll53 shall we make an issue for this to be refactored?

Reply:
Sounds good, I'll create an issue for a hotfix